### PR TITLE
API for XTools gadget, move pageviews stuff to Page

### PR DESCRIPTION
**Pull request by MusikAnimal**
Bug fix with fetching page watcher

Add more tests

Bug: https://phabricator.wikimedia.org/T170101

### CODE REVIEW

**Comment by MusikAnimal** [comment]
Note I have not written the actual JS for the gadget yet. I'd like to get this merged then work on that next, and possibly put it in version control (maybe as a .dist file).

**Comment by MusikAnimal** [comment]
Bullshit that Codecov says there's not enough coverage! With #70 I tried to get it to ignore the Repository classes (since they aren't really testable). I think if that were merged this PR would have been all green.

File: docs/api/page.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+Article info
+============
+``GET /api/articleinfo/{project}/{article}/{format}``
+
+Get basic information about the history of a page.
+
+**Parameters:**
+
+* ``project`` (**required**) - Project domain or database name.
+* ``article`` (**required**) - Full page title.
```
Comment (samwilson) [inline docs/api/page.rst:18]: Missing `format`?

**Review by Matthewrbowker — APPROVED** [review]

**Review by samwilson — APPROVED** [review]
Looks good!

**Pull request merged**